providers/common/der/der_rsa_key.c

```diff
@@ -338,8 +347,8 @@ int ossl_DER_w_RSASSA_PSS_params(WPACKET *pkt, int tag,
 
     return ossl_DER_w_begin_sequence(pkt, tag)
         && (trailerfield == default_trailerfield
-            || ossl_DER_w_ulong(pkt, 3, trailerfield))
-        && (saltlen == default_saltlen || ossl_DER_w_ulong(pkt, 2, saltlen))
+            || ossl_DER_w_uint32(pkt, 3, trailerfield))
+        && (saltlen == default_saltlen || ossl_DER_w_uint32(pkt, 2, saltlen))
```

Comment:
There is an implicit cast here from `int` to `uint32_t` (with both saltlen and trailerfield). Do we ever run on a platform where `INT_MAX > UINT32_MAX`? Perhaps there should be an explicit check for this?